lib/fluent/plugin/out_google_cloud.rb

```diff
@@ -388,6 +394,33 @@ def configure(conf)
       @resource.labels.freeze
       @common_labels.freeze
 
+      # If prometheus monitoring is enabled, register metrics in the default
+      # registry and store metric objects for future use. If metrics are
+      # registed already, just store the metric objects.
+      if @prometheus_monitoring_enabled
+        require 'prometheus/client'
+
+        prometheus = Prometheus::Client.registry
+
+        # Exception-driven behavior to avoid synchronization errors
+        begin
+          @prometheus_http_requests_count = prometheus.counter(
```

Comment:
Fair enough -- I admit I omitted the synchronization bits. But you missed my point -- I was asking to separate enabling monitoring from selecting Prometheus. I'm happy to have the `Monitoring` module be a no-op for now when `prometheus/client` can't be loaded. Then the non-Prometheus code becomes even simpler.